integration/manifests/netwatcher/0netwatcher_rbac.yaml

```diff
@@ -17,11 +17,22 @@ rules:
   - "danm.k8s.io"
   resources:
   - danmnets
+  - danmeps
   verbs:
   - get
   - list
   - watch
   - update
+  - create
+- apiGroups:
+  - ""
+  resources:
```

Comment:
I don't think either the netwatcher, or the installer needs pods, or services

Reply:
Without these permissions I get this error:
default   2019-02-14 13:51:18 +0100 CET   2019-02-14 13:51:18 +0100 CET   1         test-55678dddc4-xfwdj.15833c02ad922285   Pod                 Warning   FailedCreatePodSandBox   kubelet, node2.testdhn.local   Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "341f11a5b668ba357e933bd9c1980d5dae6e06045708db275e6fd83f25540d02" network for pod "test-55678dddc4-xfwdj": NetworkPlugin cni failed to set up pod "test-55678dddc4-xfwdj_default" network: Pod manifest could not be parsed with error: failed to get Pod info from K8s API server due to:pods "test-55678dddc4-xfwdj" is forbidden: User "system:serviceaccount:kube-system:netwatcher" cannot get resource "pods" in API group "" in the namespace "default"

Reply:
why are you using the netwatcher ServiceAccount for the DANM CNI, though?
it should use its own kubeconfig

Reply:
It gets it own kubeconfig :) 
Would you like me to create a separate serviceAccount for danm with the required roles?
Installation guide does not really specify this properly.....

Reply:
Kubeconfig and ServiceAccount are kind of mutually exclusive.
Practically speaking the log suggests the DANM CNI was initalized from InCluster config in your environment, and not from its kubeconfig.
on second thought you kind of "forced" this InCluster initialization from the kubeconfig you provided to the CNI. but you can simply state a username in your kubeconfig, and embed the certificates to be used into the kubeconf of DANM

from security POV the CNI definitely needs to have its own role, whether it is configured through InCluster config, or through embedded certs

Reply:
(own role, together with its own user)

Reply:
For a reference, our kubeconfig in the product looks liek this:
[cloudadmin@controller-1 ~]$ sudo cat /etc/kubernetes/kubeconfig/danmc.yml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: <BASE64_ENCODED_CA_CERT>
    server: https://10.254.0.1:443
  name: kubernetes
contexts:
- context:
    cluster: kubernetes
    user: danm
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: danm
  user:
    client-certificate-data: <BASE64_ENCODED_CLIENT_CERT>
    client-key-data: <BASE64_ENCODED_CLIENT_KEY>